tools/PI/DevHome.PI/Helpers/ExternalTool.cs

```diff
@@ -115,8 +96,38 @@ private async void GetIcons()
             _log.Error(ex, "Failed to get tool image");
         }
     }
+
+    public override IconElement GetIcon()
+    {
+        return new ImageIcon
+        {
+            Source = ToolIconSource,
+        };
+    }
+
+    internal async override void InvokeTool(Window? parentWindow, int? targetProcessId, HWND hWnd)
+    {
+        try
+        {
+            var process = await InvokeToolInternal(targetProcessId, hWnd);
+        }
+        catch (Exception ex)

```

Comment:
NIT: Catch `InvalidOperationException` instead because the exception `InvokeToolInternal` throws is `InvalidOperationException`

Reply:
Well, I don't want any exception to be unhandled here. Is there an advantage to only catching one type of exception?

Reply:
Readability and adding intent to the code.  Catching `InvalidOperationException` tells readers that the method knows how to deal with that particular exception.

If the method can't do anything helpful for other exceptions, it should bubble up to a try/catch in the stack.

Also, `InvokeToolExternal` already catches `Exception` so InvokeTool will never get an Exception from InvokeToolsExternal.

Reply:
The code does pop an error message letting the user know that it couldn't launch a tool. Nobody further up the stack has anything more to say about it. For errors launching a tool, the buck stops here.